grammar.js

```diff
@@ -905,6 +925,99 @@ module.exports = grammar({
       ')',
     ),
 
+    window_partition_by_expression: $ => seq(
+        $.keyword_partition_by,
+        $.identifier,
+    ),  
+
+    window_order_by_expression: $ => seq(
+        $.keyword_order_by,
+        $.identifier,
+        optional(
+            seq(
+                choice(
+                    $.direction,
+                    $.keyword_using,
+                ),
+                optional(
+                    seq(
+                        $.keyword_nulls,
+                        choice(
+                            $.keyword_first,
+                            $.keyword_last,
+                        ),
+                    ),
+                ),
+            ),
+        ),
+    ),  
+
+    _frame_definition: $ => seq(
+        choice(
+            $.keyword_unbounded_preceding,
+            seq(
+                alias($._number, $.literal),
+                $.keyword_preceding,
+            ),
+            $.keyword_current_row,
+            seq(
+                alias($._number, $.literal),
+                $.keyword_following,
+            ),
+            $.keyword_unbounded_following,
+        ),
+    ),
+
+    window_frame_clause: $ => seq(
+        choice(
+            $.keyword_range,
+            $.keyword_rows,
+            $.keyword_groups,
+        ),
+        optional(
+            $.keyword_between,
+        ),
+        $._frame_definition,
+        $.keyword_and,
```

Comment:
`and` is only required with the `between`, which to be strictly correct would probably require a choice() of seq(between, _frame_definition, and, _frame_definition) or seq(_frame_definition, _frame_definition).